### PR TITLE
Actually enforce what error message says: Protocol relative only allowed when container is HTTPS.

### LINKED ISSUE #2027: Documentation issue with amp-video
Hi, here
https://www.ampproject.org/docs/reference/amp-video.html

https is required for src, but on console log I see "https:// or // is required"

using //, everything works fine and it is validated. is https really necessary?

### PR DESCRIPTION
**Pull request by cramforce**
Fixes #2027

### CODE REVIEW

File: src/url.js [hunk 1]
```diff
@@ -92,12 +92,15 @@ export function addParamsToUrl(url, params) {
  * Provides an exception for localhost.
  * @param {string} urlString
  * @param {!Element} elementContext Element where the url was found.
+ * @param {!Window=} opt_win
  * @return {string}
  */
-export function assertHttpsUrl(urlString, elementContext) {
+export function assertHttpsUrl(urlString, elementContext, opt_win) {
```
Comment (dvoytenko) [inline src/url.js]: Will we actually ever supply `opt_win`? It seems a bit odd, given that all URLs are resolved against the current window in the line above `parseUrl(urlString)`. Or is it only for tests?

Comment (cramforce) [inline src/url.js]: Its only for tests. But your comment is correct. This was wrong in different way from what I thought.

Comment (dvoytenko) [inline src/url.js]: Are you thinking about changing this? You could also rename it to `opt_winForTestOnly` or something like that.

Comment (cramforce) [inline src/url.js]: I'll need to do something entirely different for this change :)

**Comment by cramforce** [comment]
PTAL. New fix just deletes a line.

**Comment by dvoytenko** [comment]
LGTM

**Pull request merged**